src/core/Metadata.cpp

```diff
@@ -478,6 +485,11 @@ void Metadata::setHistoryMaxSize(int value)
     set(m_data.historyMaxSize, value);
 }
 
+void Metadata::setAutosaveDelay(int value)
+{
+    set(m_data.autosaveDelay, value);
```

Comment:
Should we validate the given `value` for correctness? For example, is -1 acceptable? Should we check the upper bound, say a value as 1 year in equivalent does not make sense for this feature.
I hope the GUI part cares about it, I did not check yet the next commits.

Reply:
Fixed in [e745359](https://github.com/keepassxreboot/keepassxc/pull/9100/commits/e745359becec60f9409ed37199f6f5da3814330e)
Bounds are >=0 and <= 420000000, i used the same limit of the gui, 420000000 is just there to put a limit and avoid unexpected behavior.